File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int size=temperatures.size();
        vector<int> res(size,0);
        stack<int> ist;
        for(int index=0;index<size;index++){
            while(!ist.empty()&&temperatures[ist.top()]<temperatures[index]){
                res[ist.top()]=index -ist.top();
                ist.pop();

            }
            ist.push(index);
        }

        return res;
    }
};

File: Data Structures & Algorithms/baseball-game/submission-1.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack=[]
        res=0
        for op in operations:
            if op=="+":
                top=stack[-1]
                stack.pop()
                toAdd=top+stack[-1]
                stack.append(top)
                stack.append(toAdd)
                res+=toAdd
            elif op=="D":
                top=stack[-1]
                toAdd=top*2
                stack.append(toAdd)
                res+=toAdd
            elif op=="C":
                res-=stack[-1]
                stack.pop()
            else:
                res+=int(op)
                stack.append(int(op))
        return res
